scvi/dataset/dataset.py

```diff
@@ -1214,6 +1228,109 @@ def raw_counts_properties(
             np.squeeze(np.asarray(norm_mean2)),
         )
 
+    def highly_variable_genes(
+        self,
+        min_disp: Optional[float] = None,
+        max_disp: Optional[float] = None,
+        min_mean: Optional[float] = None,
+        max_mean: Optional[float] = None,
+        n_top_genes: Optional[int] = None,
+        n_bins: int = 20,
+        flavor: Optional[str] = "seurat",
+    ) -> pd.DataFrame:
+        """\
+        Code sample taken from the scanpy package
+        Annotate highly variable genes [Satija15]_ [Zheng17]_.
+        Depending on `flavor`, this reproduces the R-implementations of Seurat
+        [Satija15]_ and Cell Ranger [Zheng17]_.
+        The normalized dispersion is obtained by scaling with the mean and standard
+        deviation of the dispersions for genes falling into a given bin for mean
+        expression of genes. This means that for each bin of mean expression, highly
+        variable genes are selected.
+        Parameters
+        ----------
+        min_mean
+            If `n_top_genes` unequals `None`, this and all other cutoffs for the means and the
+            normalized dispersions are ignored.
+        max_mean
+            If `n_top_genes` unequals `None`, this and all other cutoffs for the means and the
+            normalized dispersions are ignored.
+        min_disp
+            If `n_top_genes` unequals `None`, this and all other cutoffs for the means and the
+            normalized dispersions are ignored.
+        max_disp
+            If `n_top_genes` unequals `None`, this and all other cutoffs for the means and the
+            normalized dispersions are ignored.
+        n_top_genes
+            Number of highly-variable genes to keep.
+        n_bins
+            Number of bins for binning the mean gene expression. Normalization is
+            done with respect to each bin. If just a single gene falls into a bin,
+            the normalized dispersion is artificially set to 1. You'll be informed
+            about this if you set `settings.verbosity = 4`.
+        flavor
+            Choose the flavor for computing normalized dispersion. In their default
+            workflows, Seurat passes the cutoffs whereas Cell Ranger passes
+            `n_top_genes`.
+        batch_correction
+            Whether batches should be taken into account during procedure
+        Returns
```

Comment:
we should clarify what this returns, since we take `.var` we can point to the scanpy docstring